conu/backend/podman/image.py

```diff
@@ -0,0 +1,412 @@
+
+        :param repository: str, see constructor
+        :param tag: str, see constructor
+        :return: instance of PodmanImage
+        """
+        if not (repository or tag):
+            raise ValueError("You need to specify either repository or tag.")
+        r = repository or self.name
+        t = tag or "latest"
+        identifier = self._id or self.get_id()
+        run_cmd(["podman", "tag", identifier, "%s:%s" % (r, t)])
+        return PodmanImage(r, tag=t)
+
+    def inspect(self, refresh=True):
+        """
+        provide metadata about the image; flip refresh=True if cached metadata are enough
+
+        :param refresh: bool, update the metadata with up to date content
+        :return: dict
+        """
+        if refresh or not self._inspect_data:
+            identifier = self._id or self.get_full_name()
+            if not identifier:
+                raise ConuException("This image does not have a valid identifier.")
+            self._inspect_data = self._inspect(identifier)
+        return self._inspect_data
+
+    @staticmethod
+    def _inspect(identifier):
+        cmdline = ['podman', 'inspect', identifier]
+        output = run_cmd(cmdline, return_output=True)
+        return json.loads(output)[0]
+
+    def rmi(self, force=False, via_name=False):
+        """
+        remove this image
+
+        :param force: bool, force removal of the image
+        :param via_name: bool, refer to the image via name, if false, refer via ID
+        :return: None
+        """
+        identifier = self.get_full_name() if via_name else (self._id or self.get_id())
+        cmdline = ["podman", "rmi", identifier, "--force" if force else ""]
+        run_cmd(cmdline)
+
+    def _run_container(self, run_command_instance, callback):
+        """ this is internal method """
+        tmpfile = os.path.join(get_backend_tmpdir(), random_tmp_filename())
+        # the cid file must not exist
+        run_command_instance.options += ["--cidfile=%s" % tmpfile]
+        logger.debug("podman command: %s" % run_command_instance)
+        response = callback()
+        # and we need to wait now; inotify would be better but is way more complicated and
+        # adds dependency
+        Probe(timeout=10, count=10, pause=0.1, fnc=lambda: self._file_not_empty(tmpfile)).run()
+        with open(tmpfile, 'r') as fd:
+            container_id = fd.read()
+        return container_id, response
+
+    @staticmethod
+    def _file_not_empty(tmpfile):
+        """
+        Returns True if file exists and it is not empty
+        to check if it is time to read container ID from cidfile
+        :param tmpfile: str, path to file
+        :return: bool, True if container id is written to the file
+        """
+        if os.path.exists(tmpfile):
+            return os.stat(tmpfile).st_size != 0
+        else:
+            return False
+
+    def run_via_binary(self, run_command_instance=None, command=None, volumes=None,
```

Comment:
I removed the --privileged flag (I'm running podman on F29, latest build)
```
18:05:20.341 __init__.py       DEBUG  command: "podman run -d --cidfile=/tmp/conu-oaoqebv1/conu-jziuqbmqhwgtboywsmcmxlswiqjftzev --label conu.test_artifact=1 24508ec0e667aa46f11d162135f4f6ab2e297f635b6edef19a8f7
5b85f5233d5 ls -lha"
18:05:20.434 __init__.py       DEBUG  container id file exists. ensure another container is not using it or delete /tmp/conu-oaoqebv1/conu-jziuqbmqhwgtboywsmcmxlswiqjftzev
```

Reply:
okay, this is a bug in podman, I'm gonna report this.

Reply:
https://github.com/containers/libpod/issues/1711 it's a known, recent issue, podman 0.11 should have a fix for this